File: Lecture 1/python_project/read_parse_files.py
class ReadParseFiles:

    def __init__(self):
        # Using name mangling to make the variable harder to access
        self.__fileFolder = "../files/"
        self.__textFileName = "text_file.txt"
        self.__jsonFileName = "json_file.json"
        self.__yamlFilename = "yaml_file.yaml"
        self.__csvFileName = "csv_file.csv"
        self.__xmlFileName = "xml_file.xml"
        self.__txtContents = None
        self.__jsonContents = None
        self.__yamlContents = None
        self.__csvContents = None
        self.__xmlContents = None

    def __call__(self):
        self.readParseTextFile()

    def fileFolder(self):
        return self.__fileFolder
    
    def textFileName(self):
        return self.__textFileName
    
    def jsonFileName(self):
        return self.__jsonFileName
    
    def yamlFileName(self):
        return self.__yamlFilename
    
    def csvFileName(self):
        return self.__csvFileName
    
    def xmlFileName(self):
        return self.__xmlFileName

    def readParseTextFile(self):
        filepath = f"{self.fileFolder()}{self.textFileName()}"

        with open(filepath, 'r') as file:
            self.__txtContents = file.read()

        print(self.__txtContents)

File: Lecture 1/node_project/readParseFiles.ts
import * as fs from 'fs';
import yamljs from 'yamljs';
import papaparse from 'papaparse';
import xml2js from 'xml2js';

export default class ReadParseFiles {

    // File info

    private readonly fileFolder: string = '../files/';
    private readonly textFilename: string = 'text_file.txt';
    private readonly jsonFilename: string = 'json_file.json';
    private readonly yamlFilename: string = 'yaml_file.yaml';
    private readonly csvFilename: string = 'csv_file.csv';
    private readonly xmlFilename: string = 'xml_file.xml';
    private readonly ENCODING = 'utf-8';

    // Data Buckets

    private txtContents: unknown;
    private jsonContents: unknown;
    private yamlContents: unknown;
    private csvContents: unknown;
    private xmlContents: unknown;

    constructor() {}

    public async readParseTextFile() {

        const filepath: string = `${this.fileFolder + this.textFilename}`; 

        try {
            const data = await fs.promises.readFile(filepath, this.ENCODING);
            this.txtContents = data;
            //console.log(this.txtContents);
        } catch (err) {
            console.error(err);
        }
    }

    public async readParseJsonFile() {

        const filepath: string = `${this.fileFolder + this.jsonFilename}`
        
        try {
            const data = await fs.promises.readFile(filepath, this.ENCODING);
            this.jsonContents = JSON.parse(data);
            //console.log(this.jsonContents);
        } catch (err) {
            console.error(err);
        }
    }

    public async readParseYamlToJson() {

        const filepath: string = this.fileFolder + this.yamlFilename;

        try {
            const data = await fs.promises.readFile(filepath, this.ENCODING);
            this.yamlContents = yamljs.parse(data.toString());
            //console.log(this.yamlContents);
        } catch (err) {
            console.error(err);
        }
    }

    public async readParseCsvToJson() {

        const filepath: string = this.fileFolder + this.csvFilename;

        try {
            const data = await fs.promises.readFile(filepath, this.ENCODING);
            this.csvContents = papaparse.parse(data, {
                header: true,
                dynamicTyping: true,
                transformHeader: header => header.toLowerCase().replace(/\W/g, '')
            }).data;
            //console.log(this.csvContents);
        } catch (err) {
            console.error(err);
        }
    }

    public async readParseXmlToJson() {

        const filepath = this.fileFolder + this.xmlFilename;
        const parser = new xml2js.Parser();

        try {
            const data = await fs.promises.readFile(filepath, this.ENCODING);
            parser.parseString(data, (err, result) => {
                if (err) {
                    console.error(err);
                    return;
                }
                try {
                    const person = result.person;

                    this.xmlContents = {
                        name: person.name[0],
                        age: parseInt(person.age[0]),
                        email: person.email[0],
                        address: {
                            street: person.address[0].street[0],
                            city: person.address[0].city[0],
                            state: person.address[0].state[0],
                            zip: person.address[0].zip[0]
                        }
                    };

                    //console.log(this.xmlContents);
                } 
                catch (err) {
                    console.error(err);
                }
            
            });
            
        } 
        catch (err) {
            console.error(err);
        }
    }
}

